staging/cse/windows/nvidiagpudriverfunc.ps1

```diff
@@ -0,0 +1,169 @@
+    $LogFolder = "C:\AzureData"
+
+    Write-Log "Attempting to install Nvidia driver..."
+
+    Write-Log "Downloading from $GpuDriverURL to $Target"
+    DownloadFileOverHttp -Url $GpuDriverURL -DestinationPath $Target -ExitCode $global:WINDOWS_CSE_ERROR_GPU_DRIVER_INSTALLATION_DOWNLOAD_FAILURE
+    Write-Log "Installer download complete"
+    
+    $IsSignatureValid = VerifySignature $Target 
+    if ($IsSignatureValid -eq $false) {
+        $ErrorMsg = "Signature embedded in $($Target) is not valid."
+        Write-Log $ErrorMsg
+        Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_INVALID_SIGNATURE -ErrorMessage $ErrorMsg
+        return
+    }
+    else {
+        Write-Log "Signature embedded in $($Target) is valid."
+    }
+
+    Write-Log "Installing $Target ..."
+    try {
+        $InstallLogFolder = "$LogFolder\NvidiaInstallLog"
+        $Arguments = "-s -n -log:$InstallLogFolder -loglevel:6"
+    
+        $p = Start-Process -FilePath $Target -ArgumentList $Arguments -PassThru
+        
+        $Timeout = 10 * 60 # in seconds. 10 minutes for timeout of the installation
+
+        # This is for testability. Start-Process mock returns a hashtable.
+        if (-not ($p -is [hashtable])) {
+            Wait-Process -InputObject $p -Timeout $Timeout -ErrorAction Stop
+        }
+    
+        # check if installation was successful
+        if ($p.ExitCode -eq 0 -or $p.ExitCode -eq 1) {
+            # 1 is issued when reboot is required after success
+            Write-Log "GPU Driver Installation Success. Code: $($p.ExitCode)"
+            Remove-InstallerFile -InstallerPath $Target
+        } 
+        else {
+            $ErrorMsg = "GPU Driver Installation Failed! Code: $($p.ExitCode)"
+            Write-Log $ErrorMsg
+            Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_GPU_DRIVER_INSTALLATION_FAILED -ErrorMessage $ErrorMsg
+            return
+        }
+
+        # check if reboot is needed
+        if ($p.ExitCode -eq 1) {
+            Write-Log "Nvidia GPU Driver Installation Finished with code 1. Reboot is needed for this GPU Driver..."
+            $global:RebootNeeded = $true
+        }
+        else {
+            Set-RebootNeeded
+        }
+    }
+    catch {
+        $ErrorMsg = "Exception: $($_.ToString())"
+        Write-Log $ErrorMsg # the status file may get over-written when the agent re-attempts this step
+        Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_GPU_DRIVER_INSTALLATION_EXCEPTION -ErrorMessage $ErrorMsg
+    }
+}
+
+function Set-RebootNeeded {
+    # check if vm size is nv series. if so, set RebootNeeded to be true
+    try {
+        $Compute = Get-VmData
+        $vmSize = $Compute.vmSize
+    }
+    catch {
+        $ErrorMsg = "Failed to query the SKU information."
+        Write-Log $ErrorMsg
+        Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_SKU_INFO_NOT_FOUND -ErrorMessage $ErrorMsg
+    }
+
+    $IsNVSeries = $vmSize -ne $null -and $vmSize -match "_NV"
```

Comment:
https://learn.microsoft.com/en-us/azure/virtual-machines/nv-series
https://learn.microsoft.com/en-us/azure/virtual-machines/nvv3-series
Could we use `Standard_NV` instead of `_NV`?